### DISCUSSION TITLE
Enhancement for the Shopping App

### DISCUSSION THREAD

**Discussion by RajGM** [the-collab-lab/tcl-60-smart-shopping-list#30] [Ideas]
Formik forms with validation and error

**Comment by RajGM** [comment]
React Hot  Toast for messages

**Comment by RajGM** [comment]
React Modal for confirmation

**Comment by labrocadabro** [comment]
For forms - personally prefer React Hook Form. But also our forms are very simple, just one text input each. Not entirely sure we need a form library for that.

I had floated the idea of using Radix UI, it's a nice headless UI component library that can be used with whatever CSS you want & it's a (hopefully) single source for UI elements. I used it recently on a personal project and found it quite easy to use. 

But am happy to go with whatever others are comfortable with.

A small UI improvement i'd like to see is click-to-copy for the token in "Use the token XXX to share your shopping list". 

One larger feature I'd really like to add if possible is managing multiple lists. I'm thinking: store an array of tokens in localstorage, in addition to the active token. Then you have a dropdown or something that shows all the tokens they've currently added and when they choose to switch to a different one, it sets that token as their currently active one. You could also have delete functionality, which would just remove the token from their localstorage, not delete the list from firebase (as other people might also have access to the same list).

**Comment by RajGM** [comment]
Instead of using localStorage, we can use context API or something else that's inside the react ecosystem.

**Comment by RajGM** [comment]
We should move everything from JSX to TS or TSX, as it's required everywhere for production.

**Comment by RajGM** [comment]
@paramsiddharth @shagunZ @shahx95 @ToobaJamal @iamkrati22 
Please contribute to this

**Comment by RajGM** [comment]
Client-side validation and test before sending a request

**Comment by RajGM** [comment]
Server-side validation and test before accepting

**Comment by RajGM** [comment]
Update convention use for react to latest standards
